frontend/client/__tests__/routes.test.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import App from '../App'
+import { MemoryRouter } from 'react-router'
+import { rootStore } from '../src/store/model'
+import NotFound from '../src/screens/404'
+import CodeValidations from '../src/screens/CodeValidations'
+import Login from '../src/screens/Login'
+import reactRouter from 'react-router-dom'
+import ChangePasswordModal from '../src/components/ChangePasswordModal'
+
+// Make sure we don't call any Firebase functions
+jest.mock('../src/store/firebase')
+
+beforeAll(() => {
+  // Mock BrowserRouter as it overrides MemoryRouter used in test
+  // Comment below suppreses eslint error that prevents us from commiting
+  // eslint-disable-next-line react/display-name
+  reactRouter.BrowserRouter = ({ children }) => <div> {children} </div>
+})
+
+test('404 for invalid link', () => {
+  const wrapper = mount(
+    <MemoryRouter initialEntries={['/random']}>
+      <App />
+    </MemoryRouter>
+  )
+  expect(wrapper.find(NotFound)).toHaveLength(1)
+})
+
+test('/ leads to /code_validations', () => {
```

Comment:
How is the user being registered as "logged in", such that this test actually works? In `'non-auth routes'`, you start by calling `rootStore.user.__update({ isSignedIn: true, isAdmin: false, isFirstTimeUser: false })` in order to mock a log in, but here you don't.

Reply:
Oh I think this is supposed to test not logged in, as it’s checking if the Login page is displayed. We should change the title of the test.